### PR TITLE
Add initial recipe for hwloc

### PR DESCRIPTION
**Pull request by inducer**

### CODE REVIEW

File: recipes/example/meta.yaml [hunk 1]
```diff
@@ -1,65 +0,0 @@
-# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (ocefpaf) [inline recipes/example/meta.yaml]: Do not remove the example! We need that :smile:

File: recipes/libhwloc/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+DISABLES="$DISABLES --disable-gl --disable-libnuma --disable-libudev"
+
+if [ `uname` == Darwin ]; then
+    ./configure --prefix=$PREFIX --disable-shared $DISABLES
+else
+    ./configure --prefix=$PREFIX $DISABLES
+fi
+
+make
+#make check
```
Comment (ocefpaf) [inline recipes/libhwloc/build.sh]: Is that failing? Can you add a comment about the failures? (We can still merge if you think the failures are not important.)

File: recipes/libhwloc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  url: https://www.open-mpi.org/software/hwloc/v1.11/downloads/{{ pname }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - gcc
```
Comment (ocefpaf) [inline recipes/libhwloc/meta.yaml]: Are you using `Fortran` or `libgomp`? If not you do not need the `gcc` package. If you do then you need to add `libgcc` or `libgfortran` (former catches all cases the latter if only `Fortran` is needed) as a run dependency.

File: recipes/libhwloc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+test:
+  commands:
+    - hwloc-ls
+
+about:
+  home: https://www.open-mpi.org/projects/hwloc/
+  license: BSD
```
Comment (ocefpaf) [inline recipes/libhwloc/meta.yaml]: Do you know which version? `BSD 3-Clause` or `BSD 2-Clause`?

File: recipes/libhwloc/build.sh (hunk 2 above)
Comment (inducer) [inline recipes/libhwloc/build.sh]: Put it back, we'll see what happens.

File: recipes/example/meta.yaml (hunk 1 above)
Comment (inducer) [inline recipes/example/meta.yaml]: Put it back.

File: recipes/libhwloc/meta.yaml (hunk 3 above)
Comment (inducer) [inline recipes/libhwloc/meta.yaml]: Removed.

File: recipes/libhwloc/meta.yaml (hunk 4 above)
Comment (inducer) [inline recipes/libhwloc/meta.yaml]: 3-Clause. Added more specific data to recipe.

File: recipes/libhwloc/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  run:
+    - libxml2
+
+test:
+  commands:
+    - hwloc-ls
+
+about:
+  home: https://www.open-mpi.org/projects/hwloc/
+  license: BSD (3-Clause)
```
Comment (ocefpaf) [inline recipes/libhwloc/meta.yaml]: No need for the parenthesis (but you don't need to change it either).

File: recipes/libhwloc/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ pname }}-{{ version }}.tar.gz
+  url: https://www.open-mpi.org/software/hwloc/v1.11/downloads/{{ pname }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - disable-failing-test.patch
```
Comment (ocefpaf) [inline recipes/libhwloc/meta.yaml]: Can you just add a comment about the patch to help us remember the reasons for it?

**Comment by ocefpaf** [comment]
Looks good just two minor comments and we can merge this. Thanks @inducer!

File: recipes/libhwloc/meta.yaml (hunk 6 above)
Comment (inducer) [inline recipes/libhwloc/meta.yaml]: Done.

File: recipes/libhwloc/meta.yaml (hunk 5 above)
Comment (inducer) [inline recipes/libhwloc/meta.yaml]: Fixed.

**Comment by inducer** [comment]
Thanks for your comments!

**Pull request merged**